crates/edr_napi/src/trace/message_trace.rs

```diff
@@ -37,15 +40,19 @@ pub struct CreateMessageTrace {
     pub depth: u32,
     // `BaseEvmMessageTrace`
     pub code: Uint8Array,
-    pub steps: Vec<Either4<EvmStep, PrecompileMessageTrace, CreateMessageTrace, CallMessageTrace>>,
-    // TODO: Will be later filled on the JS side but we should port to ContractsIdentifier in Rust
-    // This is explicitly `any` on the JS side to side-step the type-checking until we port
-    pub bytecode: Option<Value>,
+    pub steps: Vec<Either4<EvmStep, PrecompileMessageTrace, CallMessageTrace, CreateMessageTrace>>,
+    #[napi(ts_type = "Bytecode")]
+    pub bytecode: Option<Object>,
     pub number_of_subtraces: u32,
     // `CreateMessageTrace`
-    // NOTE: we can't use Option<T> as this is converted to an optional property,
-    // which is not backwards compatible with the current JS interface
-    pub deployed_contract: Either<Uint8Array, Undefined>,
+    // HACK: It seems that `Either<Uint8Array, Undefined>` means exactly what we
```

Comment:
FYI, backwards compatibility for Hardhat's internals is not a requirement, so if it's easier, you can just adopt `Option<Uint8Array>` and change the Hardhat code.